spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,260 @@
+            QMessageBox.Yes, QMessageBox.No)
+        if reply == QMessageBox.Yes:
+            self.cancelled = True
+            self.cancel_thread_install_update()
+            self.setup()
+            self.accept()
+            return True
+        return False
+
+    def continue_install(self):
+        """
+        Continue the installation in progress by downloading the installer.
+        """
+        reply = QMessageBox(icon=QMessageBox.Question,
+                            text=_('Do you want to download and'
+                                   ' install the latest version of'
+                                   ' Spyder?<br>'),
+                            parent=self._parent)
+        reply.setWindowTitle("Spyder")
+        reply.setAttribute(Qt.WA_ShowWithoutActivating)
+        reply.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+        reply.exec_()
+        if reply.result() == QMessageBox.Yes:
+            self.start_installation_update()
+        else:
+            self._change_update_installation_status(status=PENDING)
+
+    def finished_installation(self, status):
+        """Handle finished installation."""
+        if status == FINISHED or status == PENDING:
+            self.setup()
+            self.accept()
+
+    def close_installer(self):
+        """Close the installation dialog."""
+        if (self.status == FINISHED
+                or self.status == CANCELLED):
+            self.setup()
+            self.accept()
+        else:
+            self.hide()
+
+    def reject(self):
+        """Reimplemented Qt method."""
+        on_installation_widget = self._installation_widget.isVisible()
+        if on_installation_widget:
+            self.close_installer()
+        else:
+            super(UpdateInstallerDialog, self).reject()
+
+    def _change_update_installation_status(self, status=NO_STATUS):
+        """Set the installation status."""
+        self.status = status
+        self.sig_installation_status.emit(self.status)
+
+    def _progress_reporter(self, block_number, read_size, total_size):
+        progress = 0
+        if total_size > 0:
+            progress = block_number * read_size
+        if self.cancelled:
+            raise UpdateInstallationCancelledException()
+        else:
+            self.sig_download_progress.emit(progress, total_size)
+
+    def cancel_thread_install_update(self):
+        self._change_update_installation_status(status=CANCELLED)
+        self.thread_install_update.join()
+
+    def _download_install(self):
+        try:
+            with TemporaryDirectory(prefix="Spyder-") as tmpdir:
+                destination = os.path.join(tmpdir, 'updateSpyder.exe')
+                url = (
+                    'https://github.com/spyder-ide/spyder/releases/latest/'
+                    'download/Spyder_64bit_lite.exe')
+                val = (is_module_installed('numpy') or
+                       is_module_installed('pandas'))
+                if val:
```

Comment:
Lets change this variable name to something like `is_full_installer`. So something like:

```suggestion
                is_full_installer = (is_module_installed('numpy') or
                                     is_module_installed('pandas'))
                if is_full_installer:
```